src/linting/checks.jl

```diff
@@ -460,7 +460,10 @@ function check_farg_unused(x::EXPR)
         end
         if is_call(sig)
             for i = 2:length(sig)
-                if hasbinding(sig[i])
+                if is_macro_call(sig[i]) && sig[i].args[1].args[2].val == "nospecialize"
+                    # or check for `length(sig[i].args) == 4` here to treat all 1-arg macro calls like this
+                    arg = sig[i].args[3]
+                elseif hasbinding(sig[i])
```

Comment:
We want this within the `hasbinding` check (bindings are attached to the macro call in this instance), so I'd suggest:
```julia
if hasbinding(sig[i])
    if is_macro_call(sig[i]) && sig[i].args[1].args[2].val == "nospecialize" && length(sig[i]) == 4
        # or check for `length(sig[i].args) == 4` here to treat all 1-arg macro calls like this
        arg = sig[i][3]
    else
        arg = sig[i]
    end
elseif is_kwarg(sig[i]) && hasbinding(sig[i][1])
...
```